### PR TITLE
Fixed middleware routing.

### LINKED ISSUE #121: middleware does not match '/' route
assertion will fail

```lua
local server = require('http.server').new(0, 8080)
local router = require('http.router').new()
router:use(function() return {body = 'middleware', status = 200} end, {})
server:set_router(router)
server:start()
local r = http_client.get('localhost:8080/')
assert(r.status == 200)
server:stop()
```

### PR DESCRIPTION
**Pull request by Satbek**
Now middleware matches '/' path.

Closes #121

### CODE REVIEW

File: test/integration/middleware_test.lua [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        local lua_body = json.decode(resp.body)
+        lua_body.message = 'hello world!'
+        resp.body = json.encode(lua_body)
+
+        return resp
+    end
+
+    local ok = g.router:use(add_helloworld_to_response, {
+        name = 'hello_world',
+        path = '/.*',
```
Comment (printercu) [inline test/integration/middleware_test.lua:46]: Is this the same as omitting `path`? If so, let's omit it.

File: test/integration/middleware_test.lua [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        'http://127.0.0.1:12345/fruits/apple'
+    )
+    t.assert_equals(r.status, 200, 'status')
+    parsed_body = json.decode(r.body)
+    t.assert_equals(parsed_body.kind, 'orange', 'route swapped from apple handler to orange')
+end
+
+g.test_middleware_routing = function()
+    g.router:route({path = 'some_path'}, function() return {body = 'from_route', status = 200} end)
+    g.router:use(function() return {body = 'middleware', status = 200} end, {})
```
Comment (printercu) [inline test/integration/middleware_test.lua:129]: Let's make it possible to `:use(middleware)` without empty options.

File: http/router/init.lua [hunk 3]
```diff
@@ -179,7 +179,7 @@ local function use_middleware(self, handler, opts)
     opts.handler = handler
 
     local uuid = require('uuid')
-    opts.path = opts.path or '/.*'
+    opts.path = opts.path or '?.*'
```
Comment (printercu) [inline http/router/init.lua:182]: what is `?` for? Doesn't it work with `.*` or empty string?

**Comment by Totktonada** [comment]
We finally discarded http v2: https://github.com/tarantool/http/releases/tag/1.1.1.

**Pull request closed without merging**